File: lib/models/chat_message.dart
enum MessageType {
  text,
  system;

  static MessageType fromString(String value) {
    switch (value) {
      case 'text':
        return MessageType.text;
      case 'system':
        return MessageType.system;
      default:
        return MessageType.text;
    }
  }
}

class ChatMessage {
  final int id;
  final String matchId;
  final int senderId;
  final String? senderUsername;
  final String content;
  final MessageType messageType;
  final DateTime createdAt;

  ChatMessage({
    required this.id,
    required this.matchId,
    required this.senderId,
    this.senderUsername,
    required this.content,
    required this.messageType,
    required this.createdAt,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json['id'] as int,
      matchId: json['match_id'] as String,
      senderId: json['sender_id'] as int,
      senderUsername: json['sender_username'] as String?,
      content: json['content'] as String,
      messageType: MessageType.fromString(json['message_type'] as String),
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'match_id': matchId,
      'sender_id': senderId,
      'sender_username': senderUsername,
      'content': content,
      'message_type': messageType.name,
      'created_at': createdAt.toIso8601String(),
    };
  }
}

File: backend/app/models/chat_message.py
from sqlalchemy import Column, Integer, String, Text, Enum, DateTime, ForeignKey, Index
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
import enum
from app.database import Base

class MessageType(str, enum.Enum):
    text = "text"
    system = "system"

class ChatMessage(Base):
    __tablename__ = "chat_messages"

    id = Column(Integer, primary_key=True, index=True)
    match_id = Column(String(36), ForeignKey("matches.match_id", ondelete="CASCADE"), nullable=False)
    sender_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    content = Column(Text, nullable=False)
    message_type = Column(Enum(MessageType), default=MessageType.text, nullable=False)
    created_at = Column(DateTime, server_default=func.now())

    # Relationships
    sender = relationship("User")
    match = relationship("Match")

    # Indexes
    __table_args__ = (
        Index('idx_match_id', 'match_id'),
        Index('idx_created_at', 'created_at'),
    )

